### DISCUSSION TITLE
Facilitate Sales of Digital Goods

### DISCUSSION THREAD

**Discussion by SeedSigner** [btcpayserver/btcpayserver#4785] [Ideas / Feature Requests]
I first proposed this concept in the BTCPay Telegram community and was asked to post it here as well as a Feature Request. The concept is for users to be able to sell digital goods using a BTCPay instance without the need for any other infrastructure. My specific use case is to sell 3D-printable .stl models but other content creators may have an interest in selling images, videos, audio files, documents or other digital media. Ideally a user would be able to upload a digital file to the BTCPay server, set a price, and create a checkout flow that makes sense for a given item. Once buyers have selected and paid for an item, they would be presented with a one-time or expirable download link for the purchased item. I am personally excited about the v4v possibilities that this feature could open up for digital content creators.

(For context, this was a response to my question in the Telegram community: "Actually we have the primitives in place for this. We have the file system which can generate unique expirable links for each file, a receipt page that can show such links when an invoice is settled, and a pos app that has a json item template that could arbitrarily hold a file ID that gets unlocked when an invoice for it gets paid. Please open a feature request so everything gets glued together!")

**Comment by dennisreimann** [comment]
As kukks said, BTCPay has all the parts, so that it could be build already with the API. I have a paywall plugin in mind for this and it could cater to this particular use case and even cool features like using the LN invoice preimage for authenticating access.

I think replicating something like what Lightning Labs did with [LSAT](https://lsat.tech/introduction) + Aperture in a BTCPay-native way would be very useful for these kinds of use cases ­­— bringing HTTP 402 to life.
